connect/runtime/src/main/java/org/apache/kafka/connect/util/TopicAdmin.java

```diff
@@ -70,6 +74,9 @@
 public class TopicAdmin implements AutoCloseable {
 
     public static final TopicCreationResponse EMPTY_CREATION = new TopicCreationResponse(Collections.emptySet(), Collections.emptySet());
+    private static final List<Class<? extends Exception>> CAUSES_TO_RETRY_TOPIC_CREATION = Arrays.asList(
+            InvalidReplicationFactorException.class,
+            TimeoutException.class);
```

Comment:
Since TimeoutException is a RetriableException I was wondering whether we could refactor InvalidReplicationFactorException because in some circumstances it can be retriable and I think even if there won't be more brokers, it doesn't hurt too much to retry a few times with backoff. What do you think @mimaison, would this be considered an API change?

Reply:
IMO this would be a bit too drastic; the fallout of a change like that scales linearly with the retry duration that the user has configured and may delay surfacing valid issues for a bit too long. It could be especially frustrating in quickstart scenarios where someone defines a replication factor that exceeds the number of local brokers and has to wait an entire minute to find out what the problem is.